### DISCUSSION TITLE
token.sentiment is only outputting 0.0

### DISCUSSION THREAD

**Discussion by Ibrokhimsadikov** [explosion/spaCy#12547] [Help: Coding & Implementations] [answered]
in any polarity token token.sentiement attribute is giving 0.0 score, why is that?

Thank you for any suggestion

**Comment by svlandeg** [comment, marked as answer]
The `sentiment` attribute isn't being automatically filled by any spaCy components - it's there for convenience in case you want to implement sentiment analysis using a [rule-based](https://spacy.io/usage/rule-based-matching) or [machine learning](https://spacy.io/usage/training) approach (using [textcat](https://spacy.io/api/textcategorizer)). There's a simple example in the docs for a rule-based approach: https://spacy.io/usage/rule-based-matching#example3.

**Reply by Ibrokhimsadikov** [reply]
Thank you @svlandeg  for your prompt reply, but still I not sure if I understood your solution

**Reply by svlandeg** [reply]
This functionality doesn't work out-of-the box. You will have to train a sentiment analysis classifier first, and store its results in the `sentiment` attribute, for it to be useful.
